security/pkg/pki/ca/mock/fakeca.go

```diff
@@ -27,10 +27,12 @@ type FakeCA struct {
 	SignedCert    []byte
 	SignErr       *ca.Error
 	KeyCertBundle util.KeyCertBundle
+	ReceivedIDs   []string
 }
 
 // Sign returns the SignErr if SignErr is not nil, otherwise, it returns SignedCert.
-func (ca *FakeCA) Sign([]byte, []string, time.Duration, bool) ([]byte, error) {
+func (ca *FakeCA) Sign(csr []byte, identities []string, lifetime time.Duration, forCA bool) ([]byte, error) {
```

Comment:
What will the identities format look like? just service account or full SVID?

Reply:
It can be both service account or SPIFFE ID. https://github.com/istio/istio/blob/master/security/pkg/pki/util/san.go#L76 this is called to construct the correct SPIFFE ID from the identity.